### DISCUSSION TITLE
[AMA] SP-API Guard

### DISCUSSION THREAD

**Discussion by sungolivia** [amzn/selling-partner-api-models#590] [Ideas]
![logo_guard](https://github.com/amzn/selling-partner-api-docs/assets/7606170/f3fce2ac-2701-4c15-9d5d-8aec85f6d6e4)

Hey everyone, we are planning to do **our very first Ask Me Anything (AMA)** session with Amazon Selling Partner API Guard team! 🚀 

Our Product Manager, Software Development Engineers, and Security Solutions Architect from Guard team will be with us for an hour so feel free to ask away anything you had in mind relevant to SP-API Guard.
 _Feel free to leave questions in the comments beforehand, we are all very looking forward to them all._ ✏️ 

If you have not taken a look at SP-API Guard yet, check out below resources to learn more. See you then!

📆 Time: May 24th, 9-10 am PST
🌏 Location: **Virtual** (In this thread! No webinar link so please leave your questions in this thread!)

- Quick Guard intro: https://www.youtube.com/watch?v=AvATdgH1U2w
- How to use SP-API Guard: https://youtu.be/7zOt8H4fKNs
- Guard documentation: https://developer-docs.amazon.com/sp-api/docs/sp-api-guard-implementation-guide

**Comment by daapower** [comment]
I use Heroku for my application which sits on top of AWS. Would I be able to use SP-API Guard? Or do I have to directly have my application with AWS to use Guard? Is there like a Webex for this or do we just come back to this page on 5/24 9AM

**Reply by sungolivia** [reply]
Our team at Amazon will be answering on that day, so feel free to come back! No webex scheduled at the moment but would you be interested in seeing one? 👀

**Reply by daapower** [reply]
Just wasn't sure how the AMA works if there was a webcast or something. Mainly wanted to understand how it works if you aren't on AWS directly

**Reply by bchecketts** [reply]
It looks like they are doing a Reddit-style AMA where it is all done via posts on this page.  The SPAPI Guard team will reply by responding to messages

**Reply by montanagwynn** [reply]
Hello,

Thank you for your question! I am an engineer from the SP-API Guard team. SP-API Guard is available for use for applications hosted on AWS. In order to utilize Guard, you must have console access to the AWS account containing your application infrastructure. Once installed to an AWS account, it will scan resources within that account to generate a security findings report. Non-AWS resources are not currently covered by SP-API Guard scans.

Montana

**Comment by AmeerHamzaKhan97** [comment]
Where we will get the joining link?

**Reply by sungolivia** [reply]
So this will be held virtually, just like other AMAs you have seen elsewhere! Our team will be present during that time so please leave your questions as comments here! This won't be a webinar just to clarify 😄

**Comment by img-media** [comment]
I would really like to know why it's so difficult to get PII access for developers even of private applications. Once you need to create shipping labels or invoices in external applications PII is mandatory but we struggle to get accepted for about a year now. 

Amazons response is, that it's not clear why we need PII, but how can I describe better, that PII is mandatory for shipping label creation. This seems obvious to us and the input is also limited for only very short descriptions.

**Reply by zhakev0** [reply]
Hi, I am a Solutions Architect with the SP-API Guard team. Please submit a new case with clarifying details about your business use cases (around shipping fulfillment/label creation) in order to be considered for PII access. Do not re-open an old case. We require a new case to be submitted with your updated Developer Profile responses to address any missing controls or unclear use cases.

**Reply by img-media** [reply]
Unfortunately this is not helping much. The input field to describe WHY one needs access is so short, no-one can answer the asked questions AND additionally the questions have nothing to do with WHY in our case we need PII access. Strangely it seems that shipping label and invoice generation is not a valid use case for Amazon?!?

**Comment by MakeItRain** [comment]
Hello,
I have two questions:
  - I have some difficulties to fill the security form to enable access to customer address. I don't know who to contact to be able to met the needs.
  - How is it possible to get pricing informations for a batch of product? It will be helpful to adapt my pricings.
 Regards

**Comment by ericahoekerd** [comment]
What are the conditions as a company to get restricted access for the shipping information in an Amazon order within a public app on SP-API? The rejections are unclear how we can get access.

**Reply by zhakev0** [reply]
Hi, I am a Solutions Architect with the SP-API Guard team. If rejected, please do not reopen the case. Please be mindful of the areas called out in the rejection in relation to Amazon's [Acceptable Use Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=AUP&locale=en_US) and [Data Protection Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=DPP).  If you would like to appeal this decision, or you believe your responses were submitted incorrectly, or have updated your responses in line with those policies, we require a new case to be submitted with your updated Developer Profile responses to address missing controls or unclear use cases.

**Comment by dhruvesh-bista** [comment]
We understand that obtaining PII access, specifically customer details, is essential for both our Public App and Private App. As PII is mandatory for our operations, we recognise the challenges and difficulties associated with acquiring this sensitive information.

We have requested multiple time for the PII for our seller applications but requested is failed

Please ensure us with the path to get it in shortest time.

**Reply by zhakev0** [reply]
Hi, I am a Solutions Architect with the SP-API Guard team. The shortest path is to open a new case with your updated Developer profile responses to address any missing control or business use case information called out in the denial. Please be mindful of the areas of data handling and protection in relation to Amazon's [Acceptable Use Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=AUP&locale=en_US) and [Data Protection Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=DPP)

**Comment by dhruvesh-bista** [comment]
I have 2 more questions
1) Requesting brief explanation to Publish our public developer application which is in draft state..How to make it published.?
2) Is there availability to manage inbound shipments using API at Indian Marketplace? Specifically, could you provide information about the Inbound Shipment API requests for the Indian market?

**Comment by gtaylor44** [comment]
1. Why is it so hard to obtain PII access? 
2. Why do developers need to pay for a monthly Professional Seller subscription when they are not selling anything?
3. Why doesn't Amazon marketplace have a sandbox?

**Reply by sungolivia** [reply]
hey @gtaylor44!

1. 😞 
2. Good question! This is a feedback that we received quite a bit so we are definitely aware of this and hopefully can share something later!
3. Did you see this before? https://developer-docs.amazon.com/sp-api/docs/the-selling-partner-api-sandbox

**Reply by zhakev0** [reply]
Hi @gtaylor44, Amazon maintains the [Acceptable Use Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=AUP&locale=en_US) and [Data Protection Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=DPP) as the guiding requirements for PII access, in order to balance security with the availability of data that Developers need in order to maintain a good customer experience for Selling Partners. Thanks for your patience and collaboration.

**Comment by daapower** [comment]
Almost all of the questions that are popping up are unrelated to SP-API Guard. This isn't the general SP-API team, this is specifically the SP-API GUARD team.

**Reply by sungolivia** [reply]
thanks @daapower for the lovely reminder 👏

**Comment by RaffyKumruyan** [comment]
As many colleagues mention already, there is a struggle with getting a specific answer when applying for PII access for a Public app. We recently got this reply that is also not giving us any specific or even helpful insights on what we should change when answering the questions, our architecture, or the public website: 

```
We have reviewed the information you provided and determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time.

We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies.
```

My questions regarding the SP-API GUARD are:

1. If we run a scan using the SP-API GUARD, will we get any other more specific answers when re-applying for the PII access on a Public app?

2. Is the SP-API GUARD a tool that should only be used only after we receive a positive reply from the Developer Registration Team that we should perform an audit of the architecture?

3. If we use it right now, will it benefit us or help us anyhow with the application?

4. Are the SP-API GUARD team and Developer Registration Team connected?

5. Can you give us the contact email of someone from the Developer Registration Team?

Thanks!

**Reply by cjding** [reply]
Hi! My name is Chenjie and I'm a product manager on the SP-API Guard team. Thanks for your questions. Regarding your questions about Guard, please see my answers below.
1. Currently no. Guard enables developers to assess security controls as self-service. If you receive a security audit, you can get an approximately 30% reduction in the number of questions in the audit framework (similar in time saving) if submitting a Guard scan report. 
2. If you are currently an SP-API developer, Guard helps you check your compliance status with Amazon policy. If you are not yet an SP-API developer, you will need to pass the registration process first. However, we are actively considering using Guard for part of the registration process as well. Your thoughts and suggestions are welcomed.
3. As mentioned above, not currently, but we are actively considering.
4. We both belong to the bigger organization that manages the API system. 
5. The best way to contact the registration is through your existing application case or open a new Developer Support case. https://developer.amazonservices.com/support

**Reply by RaffyKumruyan** [reply]
Thanks for the answers @cjding.

I want to follow up on question 5. - We have tried many times to answer an existing application case or open a new one, but the replies are coming from the Amazon Support team. They always say that the Amazon Support team is not involved in the evaluation process and we need to re-apply again so the Developer Registration Team can review our application for a Public app with PII.

That is why I am asking for direct contact with someone from the Developer Registration Team, since by opening a new Developer Support case my question goes to the Amazon Support team and they do not have the authority to answer.

Thanks!

**Reply by zhakev0** [reply]
Hi @RaffyKumruyan , I am a Solutions Architect with the SP-API Guard team. Thanks for your patience, the original case closures may not have been clear on the next steps, this process has been expanded on internally. Adding to Chenjie's point no.5 above - the shortest path is to open a new case with your updated Developer profile responses to address any missing control or business use case information called out in the denial. Please be mindful of the areas of data handling and protection in relation to Amazon's [Acceptable Use Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=AUP&locale=en_US) and [Data Protection Policy](https://sellercentral.amazon.com/mws/static/policy?documentType=DPP)

**Reply by RaffyKumruyan** [reply]
Thanks @zhakev0.

"Thanks for your patience, the original case closures may not have been clear on the next steps, this process has been expanded on internally."

Can you tell me more about how the process is expanded on internally and how can I get more information for a case? Of course, expect for opening a new one.
My only goal is to get some more information from the Developer Registration team on why we are getting rejected and if internally they have it, it will be great if I can understand it as well.

Thanks!

**Reply by zhakev0** [reply]
Hi @RaffyKumruyan to clarify, the denial message content is more detailed now.  If you submit a new case and get rejected, it will point you to the specific domains in policy where there is a gap, that must be addressed in order to move forward.

**Reply by RaffyKumruyan** [reply]
Hi @zhakev0, we submitted a request for a Public app with PII last week and got the answer I pasted at this thread's beginning, which is not pointing to specific questions.

A month ago we were working on an application that needed a Private app with PII and as a response, we got specific instructions about which question exactly is wrong and what it needs to be changed there.

If you say that now the process will look like that if we reapply for the Public app now, we will do that.

I was just left with the impression that the replies for a Private app with PII are specific but for a Public app with PII are general (at least in our case and in many other peoples'). Is there something like that?

**Reply by zhakev0** [reply]
The partner team who reviews the case implemented this fairly recently. Please try to resubmit a public app case again to get more specific feedback.

**Reply by RaffyKumruyan** [reply]
Thanks @zhakev0. Will do that and follow up on the topic.

**Comment by dothework** [comment]
Good morning. We are creating a Private app that needs to fetch the order total given a order number. We have not been able to get the IAM set up correctly. We attempted 7 times and asked for help. The help has never resulted in a solution. We ALWAYS get a 403 when attempting to use the SP-API to get order details. We URGENTLY need detailed, correct, up-to-date, clear and concise instructions on how to setup a IAM Role (or User) for SP-API use. 

Can you post that here? Please. Thank you.

**Reply by sungolivia** [reply]
Hey @dothework even though this is only for SP-API Guard related, here are some resources to help you out!
https://developer-docs.amazon.com/sp-api-blog/docs/troubleshooting-selling-partner-api-authorization-errors
https://developer-docs.amazon.com/sp-api/docs/sp-api-errors-frequently-asked-questions

I would also just post on our Discussion tab so it can be addressed as well!

**Comment by bchecketts** [comment]
Is is possible to estimate the cost of running SP-API Guard?

It would help to understand some more details about what it actually does.

Is it reading through all of my S3 buckets?  My databases? That could be many terabytes of information that is potentially costly, and honestly a little scary to grant that type of permission to something I don't fully understand

**Reply by Madhusudhan441** [reply]
Hello,

I'm an Engineer on the SP-API Guard team. Thank you for your question. There is a cost cap of $500 when running SP-API Guard, which Guard will not exceed. You may reference the Cost breakdown for further estimations on pricing: https://developer-docs.amazon.com/sp-api/docs/sp-api-guard-cost.
Guard performs Sampling based approach on S3 Buckets and RDS Databases. This significantly reduces the scope of scan from terabytes of data to gigabytes. This is already taken into consideration in our cost estimation.

**Comment by sungolivia** [comment]
Hello everyone! Thanks for coming out! We are on our way to reading through questions and typing away!

**Comment by lapkritinis** [comment]
My experience with this tool has been characterized by significant disappointment due to at least two major problems.

The first issue is that upon activation, the tool introduces new findings, such as EC2 instances with public IP addresses, VPCs with no flow logs, among others. These findings, which are a direct result of launching this product, end up in the report, thereby complicating its interpretation.

The second issue lies in the tool's method of operation. It sets up and employs Config and Security Hub for scanning, and subsequently disables them. While this might seem appropriate at first glance, it leads to further complications. Specifically, each new scan generates new Config rules with random suffixes. Subsequently, these rules result in findings within the Security Hub. After the scan is completed, these services are disabled. However, in the event of subsequent scans, the old findings in the Security Hub remain, even though they now have broken links to the actual Config rules. This results in persistent findings in the Security Hub, with a "last checked date" that is never updated.

**Reply by sungolivia** [reply]
hey @lapkritinis  we are glad that you had a chance to try Guard out, and really appreciate the feedback. If there is something else that you would like to see improved, can you tell us more about it?

**Reply by lapkritinis** [reply]
The two identified areas are indeed of primary importance. I have recently activated both the Config and Security Hub services and am currently navigating through the findings in real-time, so to speak. However, there remains the imperative task of addressing and rectifying the previously identified discrepancies or "broken findings". Ultimately, my key aspiration is to have a streamlined method to rerun audits and obtain the most current findings that persist. This would ensure that we are addressing all the outstanding issues in our system effectively.

**Reply by Madhusudhan441** [reply]
Hello, I am an engineer from the SP-API Guard team. Thank you for the feedback. We recently took an initial stab at the first issue and addressed in a recent fix. That fix should mitigate few of those resources in the report. On the other one, Guard currently leverages those AWS Security Services behind the hood. We will reach out to appropriate team to mitigate the issue.

**Comment by daapower** [comment]
How does one integrate this with Heroku that is running on AWS and using postgresql that is also hosted on Heroku that sits on AWS. Heroku controls the AWS instances.

Also, does SP-API guard see my database tables?

**Reply by montanagwynn** [reply]
Hello,

Thank you for your question! I am an engineer from the SP-API Guard team. SP-API Guard is available for use for applications hosted on AWS. In order to utilize Guard, you must have console access to the AWS account containing your application infrastructure. Once installed to an AWS account, it will scan resources within that account to generate a security findings report. Non-AWS resources are not currently covered by SP-API Guard scans.

Montana

**Reply by montanagwynn** [reply]
SP-API Guard will scan over your database tables as a part of generating the security report via a sampling-based approach. Data from your database tables will not be shared with Amazon.

Montana

**Reply by sungolivia** [reply]
Also you choose to share your findings with Amazon so check out the video here at the timestamp where you do that!
https://youtu.be/7zOt8H4fKNs?t=129

**Comment by bchecketts** [comment]
Can you enumerate which AWS services that Guard is examining. I see you mentioned S3 and RDS here.  How about DynamoDB?  Step Functions? Redshift?

**Reply by Madhusudhan441** [reply]
Hello, 
I am an engineer from the SP-API Guard team. Thank you for your question. Currently, Guard does not support scanning other Database resources than S3 and RDS. Guard also supports few other resources such as Amazon EC2 instances, Amazon ECS, and Amazon EKS. You can find more information at https://developer-docs.amazon.com/sp-api/docs/sp-api-guard-components.

**Comment by sungolivia** [comment]
Thank you for participating everyone and the lovely feedback as well, if you weren't here but still have questions, please feel free to drop them and I will ping our SP-API Guard to come back 😆 

On the other hand, if there is any other topic for AMA, let me know!

**Comment by lapkritinis** [comment]
Dear @sungolivia,

I've encountered an issue with the GUARD_ALB_WAF_ENABLED rule that I'm hoping you could help with.

Despite enabling both WAFv2 and WAF Classic, I'm persistently receiving a "non-compliant" status. I'm seeing these messages:

WebACLId not found in the rule parameters: 44a3a9eb-ca39-4c87-846a-61458bccd835
WebACLId not found in the rule parameters: arn:aws:wafv2:eu-central-1:123456789012:regional/webacl/waf-default-prod/3e266f69-9919-48d2-b40a-3fdbd1322ece
Interestingly, these error messages persist whether I use WAF Classic or WAFv2. Could you kindly provide some insight?

**Reply by cjding** [reply]
Hi! I'm a product manager with SP-API Guard. My team would like to understand this issue better in order to help resolve it. Have you created a support case on this? If you can copy this description to a support case and let us know your Developer ID in the case, we would like to take it from there and find the right team member to assist you. Please let me know!

**Reply by lapkritinis** [reply]
A support case with AWS?
I am finishing with all the other findings - only one left and I will rerun a fresh scan with the freshly launched template. If that rule will be still there and will be failing - I will open a case. How do I share the developer ID privately?

**Comment by lapkritinis** [comment]
I have cleared all findings. Then I installed Guard. It generated 14 new findings in Security Hub. Presenting them for reference:

```
"Compliance Status","Severity","ID","Title","Failed checks","Unknown checks","Not available checks","Passed checks"
"Failed","Low","CloudFormation.1","CloudFormation stacks should be integrated with Simple Notification Service (SNS)","1","0","0","0"
"Failed","High","EC2.2","The VPC default security group should not allow inbound and outbound traffic","1","0","0","3"
"Failed","Medium","EC2.6","VPC flow logging should be enabled in all VPCs","1","0","0","3"
"Failed","High","EC2.8","EC2 instances should use Instance Metadata Service Version 2 (IMDSv2)","1","0","0","6"
"Failed","High","EC2.9","EC2 instances should not have a public IPv4 address","1","0","0","6"
"Failed","Medium","EC2.10","Amazon EC2 should be configured to use VPC endpoints that are created for the Amazon EC2 service","1","0","0","3"
"Failed","Medium","EC2.21","Network ACLs should not allow ingress from 0.0.0.0/0 to port 22 or port 3389","1","0","0","12"
"Failed","Medium","KMS.2","IAM principals should not have IAM inline policies that allow decryption actions on all KMS keys","2","0","0","89"
"Failed","Medium","S3.5","S3 buckets should require requests to use Secure Socket Layer","1","0","0","7"
"Failed","Medium","S3.9","S3 bucket server access logging should be enabled","1","0","0","6"
"Failed","Medium","S3.11","S3 buckets should have event notifications enabled","1","0","0","7"
"Failed","Low","S3.13","S3 buckets should have lifecycle policies configured","1","0","0","7"
"Failed","Medium","SNS.1","SNS topics should be encrypted at-rest using AWS KMS","2","0","0","2"
"Failed","Medium","SNS.2","Logging of delivery status should be enabled for notification messages sent to a topic","2","0","0","2"
```
